walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,217 @@
+      scheduleIterator.remove();
+      // If the event is contained within the free slot, shorten the slot
+    } else if (freeTime.contains(event.getWhen())) {
+      scheduleIterator.remove();
+      if (freeTime.start() != event.getWhen().start()) {
+        scheduleIterator.add(
+            TimeRange.fromStartEnd(freeTime.start(), event.getWhen().start(), false));
+      }
+      if (freeTime.end() != event.getWhen().end()) {
+        scheduleIterator.add(TimeRange.fromStartEnd(event.getWhen().end(), freeTime.end(), false));
+      }
+      // If the event overlaps with the free slot, shorten the slot based on whether the start or
+      // end overlaps.
+    } else if (freeTime.overlaps(event.getWhen())) {
+      scheduleIterator.remove();
+      if (freeTime.start() < event.getWhen().end()) {
+        scheduleIterator.add(TimeRange.fromStartEnd(event.getWhen().end(), freeTime.end(), false));
+      } else if (freeTime.end() > event.getWhen().start()) {
+        scheduleIterator.add(
+            TimeRange.fromStartEnd(freeTime.start(), event.getWhen().start(), false));
+      }
+    }
+    return schedules;
+  }
+
+  private ArrayList<TimeRange> determineAvailableTime(MeetingRequest request, int firstFlag,
+      ArrayList<TimeRange> meetingTimesToAdd, Map<String, ArrayList<TimeRange>> schedules,
+      String attendee, ArrayList<TimeRange> meetingTimes) {
+    // Finds slot(s) where attendee is available in relation to currently available meetingTimes.
+    if (firstFlag == 1) {
+      // If first attendee, simply insert all free slots.
+      for (TimeRange free : schedules.get(attendee)) {
+        if (free.duration() >= request.getDuration()) {
+          meetingTimesToAdd.add(free);
+        }
+      }
+      // If subsequent attendee, find overlapping time slots and insert the most conservative time
+      // slot.
+    } else {
+      for (ListIterator meetingTimesIterator = meetingTimes.listIterator();
+           meetingTimesIterator.hasNext();) {
+        TimeRange firstFree = (TimeRange) meetingTimesIterator.next();
+        for (TimeRange nextFree : schedules.get(attendee)) {
+          if (nextFree.overlaps(firstFree)) {
+            TimeRange toAdd = TimeRange.fromStartEnd(Math.max(firstFree.start(), nextFree.start()),
+                Math.min(firstFree.end(), nextFree.end()), false);
+            if (toAdd.duration() >= request.getDuration()) {
+              meetingTimesToAdd.add(toAdd);
+            }
+          }
+        }
+        meetingTimesIterator.remove();
+      }
+    }
+    return meetingTimesToAdd;
+  }
+
+  private Map<TimeRange, Integer> determineFrequency(Map<TimeRange, Integer> freeCounter,
+      Map<String, ArrayList<TimeRange>> optionalFrees, MeetingRequest request) {
+    // Find how many optionals can attend each free slot
+    for (String first : optionalFrees.keySet()) {
+      for (String second : optionalFrees.keySet()) {
+        if (first != second) {
```

Comment:
You can reduce the code indention by changing this to:

if (first == second) continue;